internal/handlers/common/params.go

```diff
@@ -65,3 +67,24 @@ func AssertType[T types.Type](value any) (T, error) {
 
 	return res, nil
 }
+
+// GetInt64Value matches value's type returning protocol error for bad float values and unexpected results.
+// valueType parameter used to generate error message.
+func GetInt64Value(valueType string, value any) (int64, error) {
```

Comment:
MongoDB calls it "number" in several places. Let's use that name somehow. For example:
```
// GetNumberParam checks if the given value is a whole number and returns it as int64.
// key is used in error messages if the value is not right.
func GetNumberParam(value any, key string) (int64, error)
```

Then the implementation should be cleaned from any references to "sort" as it is not only about sorting.

We should use that function at filterFieldExprSize. And we should move a TODO item about negative zero from that function to the new one.

Reply:
> We should use that function at filterFieldExprSize.

That wasn't done